src/sentry/discover/endpoints/discover_saved_queries.py

```diff
@@ -45,8 +44,32 @@ def get(self, request, organization):
                 else:
                     queryset = queryset.none()
 
-        saved_queries = list(queryset.all())
-        return Response(serialize(saved_queries), status=200)
+        sort_by = request.query_params.get("sortBy")
+        if sort_by in ("name", "-name"):
+            order_by = sort_by
+            paginator_cls = OffsetPaginator
+        elif sort_by in ("dateCreated", "-dateCreated"):
+            order_by = "-date_created" if sort_by.startswith("-") else "date_created"
+            paginator_cls = DateTimePaginator
+        elif sort_by in ("dateUpdated", "-dateUpdated"):
+            order_by = "-date_updated" if sort_by.startswith("-") else "date_updated"
+            paginator_cls = DateTimePaginator
+        else:
+            order_by = "name"
+            paginator_cls = OffsetPaginator
+
+        # Old discover expects all queries and uses this parameter.
+        if request.query_params.get("all") == "1":
```

Comment:
@markstory  To clarify, am I correct that the `all` query param can still be applied to Discover2+ queries whenever `query` is `'version:2'`?

Reply:
Yes. If you want to get all the queries and not paginate one could send this parameter.